core/src/test/java/org/apache/iceberg/io/inmemory/InMemoryCatalogTest.java

```diff
@@ -0,0 +1,180 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io.inmemory;
+
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.hadoop.CommonCatalogTests;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.types.Types.IntegerType;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests specific to {@link InMemoryCatalog}, e.g. rename table, alter namespace properties, etc.
+ * Common catalog tests can be found in {@link CommonCatalogTests}.
+ */
+public class InMemoryCatalogTest {
+
+  private InMemoryCatalog catalog;
+
+  @Before
+  public void before() {
+    catalog = new InMemoryCatalog();
+    catalog.initialize("in-memory-catalog", ImmutableMap.of());
+  }
+
+  @Test
+  public void testSetProperties() {
+    Namespace namespace = Namespace.of("a");
+
+    // Create a namespace with properties: {'k1': 'v1', 'k2': 'v2'}
+    catalog.createNamespace(namespace, ImmutableMap.of("k1", "v1", "k2", "v2"));
+    // Verify the properties
+    assertEquals(ImmutableMap.of("k1", "v1", "k2", "v2"), catalog.loadNamespaceMetadata(namespace));
+
+    // Set properties such that 'k1' is overwritten and a new key 'k3' is added.
+    catalog.setProperties(namespace, ImmutableMap.of("k1", "v1'", "k3", "v3"));
+    // Verify the result of set properties.
+    assertEquals(ImmutableMap.of("k1", "v1'", "k2", "v2", "k3", "v3"), catalog.loadNamespaceMetadata(namespace));
+  }
+
+  @Test
+  public void testRemoveProperties() {
+    Namespace namespace = Namespace.of("a");
+
+    // Create a namespace with properties: {'k1': 'v1', 'k2': 'v2'}
+    catalog.createNamespace(namespace, ImmutableMap.of("k1", "v1", "k2", "v2"));
+    // Verify the properties
+    assertEquals(ImmutableMap.of("k1", "v1", "k2", "v2"), catalog.loadNamespaceMetadata(namespace));
+
+    // Remove properties 'k1'
+    catalog.removeProperties(namespace, ImmutableSet.of("k1", "k3"));
+    // Verify the result of set properties.
```

Comment:
Nit: These comments don't add much value in my opinion. The code is readable as is, especially given the method names. I would remove most of them.